cosmos-hive-auth-provider/src/main/java/com/telefonica/iot/cosmos/hive/authprovider/OAuth2AuthenticationProviderImpl.java

```diff
@@ -39,7 +39,8 @@
  */
 public class OAuth2AuthenticationProviderImpl implements PasswdAuthenticationProvider {
     
-    private static final Logger LOGGER = Logger.getLogger(HttpClientFactory.class);
+    private static final Logger LOGGER = Logger.getLogger(OAuth2AuthenticationProviderImpl.class);
+    private static final OAuth2Cache CACHE = new OAuth2Cache("/home/hive/oauth2.cache");
```

Comment:
I thought about that while coding this. The problem is the only of way of maintaining an object (the cache) between different calls to the `OAuth2Cache` class is through a static final object like `CACHE`. And such kind of object can only be initialized when declared, i.e. cannot be initialized in the constructor (where a parameterized file path could be read, for sure).